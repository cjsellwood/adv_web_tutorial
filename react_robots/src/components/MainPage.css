@font-face {
  font-display: swap;
  font-family: "SEGA LOGO FONT";
  font-style: normal;
  font-weight: normal;
  src: local("SEGA LOGO FONT"), url("SEGA.woff") format("woff");
}

h1 {
  font-family: "SEGA LOGO FONT";
  font-weight: 200;
  color: cyan;
}

img {
  width: 200px;
  height: 200px;
}