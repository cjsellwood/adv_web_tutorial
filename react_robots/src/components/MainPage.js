import React, { useEffect } from "react";
import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";
import Scroll from "../components/Scroll";
import ErrorBoundary from "../components/ErrorBoundary";
import "./MainPage.css";
import Header from "../components/Header";

const MainPage = ({
  searchField,
  onSearchChange,
  robots,
  isPending,
  onRequestRobots,
}) => {
  useEffect(() => {
    onRequestRobots();
  }, [onRequestRobots]);

  const filterRobots = () => {
    return robots.filter((robot) => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    });
  };

  return isPending ? (
    <h1>Loading</h1>
  ) : (
    <div className="tc">
      <Header />
      <SearchBox searchChange={onSearchChange} />
      <Scroll>
        <ErrorBoundary>
          <CardList robots={filterRobots()} />
        </ErrorBoundary>
      </Scroll>
    </div>
  );
};

export default MainPage;
