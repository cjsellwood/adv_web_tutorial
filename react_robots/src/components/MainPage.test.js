import { render, screen } from "@testing-library/react";
import MainPage from "../components/MainPage";

let mockProps;
beforeEach(() => {
  mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false,
  };
});

test("render MainPage without crashing", () => {
  expect(render(<MainPage {...mockProps} />)).toMatchSnapshot();
});

test("filters robots correctly", () => {
  mockProps = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: "John",
        email: "john@gmail.com",
      },
    ],
    searchField: "",
    isPending: false,
  };
  render(<MainPage {...mockProps} />);

  expect(screen.queryByText("John")).not.toBe(null);
});

test("filters robots correctly 2", () => {
  mockProps = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: "John",
        email: "john@gmail.com",
      },
    ],
    searchField: "a",
    isPending: false,
  };
  const filteredRobots = [];

  render(<MainPage {...mockProps} />);

  expect(screen.queryByText("John")).toBe(null);
});