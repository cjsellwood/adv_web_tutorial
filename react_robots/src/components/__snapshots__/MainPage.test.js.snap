// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MainPage without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tc"
      >
        <div>
          <h1
            class="f1"
          >
            RoboFriends
          </h1>
          <button
            color="red"
            data-testid="counter"
          >
            Count 
            0
          </button>
        </div>
        <div
          class="pa2"
        >
          <input
            aria-label="search robots"
            class="pa3 ba b--green bg-lightest-blue"
            placeholder="search robots"
            type="search"
            value=""
          />
        </div>
        <div
          style="overflow-y: scroll; border: 5px solid black; height: 500px;"
        >
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tc"
    >
      <div>
        <h1
          class="f1"
        >
          RoboFriends
        </h1>
        <button
          color="red"
          data-testid="counter"
        >
          Count 
          0
        </button>
      </div>
      <div
        class="pa2"
      >
        <input
          aria-label="search robots"
          class="pa3 ba b--green bg-lightest-blue"
          placeholder="search robots"
          type="search"
          value=""
        />
      </div>
      <div
        style="overflow-y: scroll; border: 5px solid black; height: 500px;"
      >
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
